### DISCUSSION TITLE
Package dependency resolution diagnostics and the impact on VSCode plugin experience

### DISCUSSION THREAD

**Discussion by sameerajayasoma** [ballerina-platform/ballerina-lang#32685] [General]
Currently, many package dependency resolution-related errors turn into `ProjectExcepton` instances. This behavior hurts VSCode plugin users because the compilation does not continue after such exceptions.

So I am working on converting those exceptions into package resolution diagnostics. Most of those diagnostics do not have associated locations. I.e., It doesn't make sense to map such a diagnostics with a location in the `.bal` source file. 

I had a look at how Rust handles this. It seems the Rust VSCode plugin that I use marks the whole `Cargo.toml` file as the location. I at least expected it to mark just the `[package]` because such errors related to the whole package.

<img width="1046" alt="Screen Shot 2021-09-10 at 4 40 08 PM" src="https://user-images.githubusercontent.com/2637328/132928905-a2e7104b-063f-4673-b375-ee651d721a1a.png">

Can we do something similar in Ballerina as well? 
@nadeeshaan @mohanvive @azinneera @pramodya1994 FYI.

**Comment by nadeeshaan** [comment]
@sameerajayasoma yes definitely we can do that. Also, it does not matter whether it is a `.bal` source or any other source in the workspace. What we have to get is only the document's path and the location.
